src/components/multi_select_state.js

```diff
@@ -32,20 +33,69 @@ const withMultiSelectState = WrappedComponent =>
       }
     }
 
-    selectItem(id) {
-      const { items } = this.props;
-      const { selectedItems } = this.state;
-      if (!selectedItems.find(item => item.id === id)) {
-        const newSelectedItems = items.filter(
-          item =>
-            item.id === id ||
-            selectedItems.find(selectedItem => item.id === selectedItem.id)
-        );
-        this.setState({ selectedItems: newSelectedItems }, this.handleChange);
-      } else {
-        this.unselectItem(id);
+      handleMultiSelection(id) {
+        const { items } = this.props;
+          const { selectedItems, firstItemShiftSelected, filteredItems } = this.state;
+          let newSelectedItems;
+          if(firstItemShiftSelected == null) {
+              this.setState({ firstItemShiftSelected: id });
+              newSelectedItems = items.filter(
+                  item =>
+                      item.id === id ||
+                      selectedItems.find(selectedItem => item.id === selectedItem.id)
+              );
+          } else {
+              if (id > firstItemShiftSelected) {
+                  newSelectedItems = items.filter(item =>
+                      item.id >= firstItemShiftSelected &&
+                      item.id <= id &&
+                      filteredItems.find(selectedItem => item.id === selectedItem.id) ||
+                      selectedItems.find(selectedItem => item.id === selectedItem.id)
+                  );
+              } else {
+                  newSelectedItems = items.filter(item =>
+                      item.id <= firstItemShiftSelected &&
+                      item.id >= id &&
+                      filteredItems.find(selectedItem => item.id === selectedItem.id) ||
+                      selectedItems.find(selectedItem => item.id === selectedItem.id)
+                  );
+              }
+          }
+          this.setState({selectedItems: newSelectedItems}, this.handleChange);
+      }
+
+      componentDidMount() {
+          ReactDOM.findDOMNode(this).addEventListener('keyup', this.onKeyUp);
+      }
+
+      componentWillUnmount() {
+          ReactDOM.findDOMNode(this).removeEventListener('keyup', this.onKeyUp, false);
+      }
+
+      onKeyUp(event) {
+        if(event.keyCode === 16 && !event.shiftKey) {
```

Comment:
what does it mean event.keyCode === 16 && !event.shiftKey?
isn't keyCode 16 === "shift"?